server/monitor-types/mqtt.js

```diff
@@ -48,6 +49,19 @@ class MqttMonitorType extends MonitorType {
         }
     }
 
+    /**
+     * This function checks if the actual MQTT topic matches the subscribed topic.
+     * To handle MQTT wildcards, it converts the subscribed topic into a regex pattern.
+     * @param {string} subcribedTopic MQTT subscribed topic
+     * @returns {RegExp} RegExp if the actual topic matches the subscribed topic
+     */
+    static mqttTopicRegex(subcribedTopic) {
+        subcribedTopic = subcribedTopic.replace(/([$.|?*{}()\[\]\\])/g, '\\$1'); // Escape special regex chars except + and #
+        subcribedTopic = subcribedTopic.replace(/\+/g, '[^/]+'); // Replace + with regex for one or more characters except slash
```

Comment:
I don't think this quite works when the start of the string is "+", or am I misunderstanding the syntax here wrong?